src/commands/hooks/clone_hooks.rs

```diff
@@ -0,0 +1,261 @@
+use crate::git::cli_parser::ParsedGitInvocation;
+use crate::git::repository::find_repository_in_path;
+use crate::git::sync_authorship::fetch_authorship_notes;
+use crate::utils::debug_log;
+
+pub fn post_clone_hook(
+    parsed_args: &ParsedGitInvocation,
+    exit_status: std::process::ExitStatus,
+) {
+    // Only run if clone succeeded
+    if !exit_status.success() {
+        return;
+    }
+
+    // Extract the target directory from clone arguments
+    let target_dir = match extract_clone_target_directory(&parsed_args.command_args) {
+        Some(dir) => dir,
+        None => {
+            debug_log("failed to extract target directory from clone command; skipping authorship fetch");
+            return;
+        }
+    };
+
+    debug_log(&format!(
+        "post-clone: attempting to fetch authorship notes for cloned repository at: {}",
+        target_dir
+    ));
+
+    // Open the newly cloned repository
+    let repository = match find_repository_in_path(&target_dir) {
+        Ok(repo) => repo,
+        Err(e) => {
+            debug_log(&format!(
+                "failed to open cloned repository at {}: {}; skipping authorship fetch",
+                target_dir, e
+            ));
+            return;
+        }
+    };
+
+    // Fetch authorship notes from origin
+    if let Err(e) = fetch_authorship_notes(&repository, "origin") {
```

Comment:
This could take a while on big repos, is there any way that we could background this by default on interactive (user-facing) terminals?

Reply:
```
aidancunniffe@ac Desktop % rm -rf mom-test && git clone https://github.com/acunniffe/mom-test.git

Cloning into 'mom-test'...
remote: Enumerating objects: 485, done.
remote: Counting objects: 100% (485/485), done.
remote: Compressing objects: 100% (322/322), done.
remote: Total 485 (delta 243), reused 372 (delta 130), pack-reused 0 (from 0)
Receiving objects: 100% (485/485), 15.37 MiB | 27.95 MiB/s, done.
Resolving deltas: 100% (243/243), done.
Fetching git-ai authorship notes, done.
```

Did this ⬆️